routing/http/client/client.go

```diff
@@ -79,12 +82,18 @@ func WithIdentity(identity crypto.PrivKey) Option {
 	}
 }
 
+// WithHTTPClient sets a custom HTTP Client to be used with [Client]. If you use
+// this option, [WithUserAgent] will no longer work, as it is made to work with
+// the default HTTP Client.
 func WithHTTPClient(h httpClient) Option {
 	return func(c *Client) {
 		c.httpClient = h
 	}
 }
 
+// WithUserAgent sets a custom user agent to use with the HTTP Client. This will
+// only work if you use the default HTTP Client. If you use [WithHTTPClient], you
```

Comment:
Most of the time we use options the options are allowed to return errors, for some reason that isn't the case here. Given there likely aren't (m)any options implemented not in this package WDYT about changing Option to return errors and then having `WithUserAgent` error rather than failing silently if the types are wrong?